go/vt/vtgate/vindexes/binary.go

```diff
@@ -0,0 +1,60 @@
+package vindexes
+
+import (
+	"bytes"
+	"fmt"
+)
+
+// Binary is a vindex that hashes binary bits to a keyspace id.
+type Binary struct {
+	name string
+}
+
+// NewBinary creates a new Binary.
+func NewBinary(name string, _ map[string]string) (Vindex, error) {
+	return &Binary{name: name}, nil
+}
+
+// String returns the name of the vindex.
+func (vind *Binary) String() string {
+	return vind.name
+}
+
+// Cost returns the cost as 1.
```

Comment:
I think it's better if the cost is 0. A cost of 1 is for vindexes that perform some kind of transformation (see Numeric).